### PR TITLE
chore: upgrade examples to next 15 and fix e2e test

### PR DESCRIPTION
**Pull request by sommeeeer**
This PR will be in draft mode. Trying to update all the apps in `/examples` to next 15 and fix the e2e test.

### CODE REVIEW

**Comment by sommeeeer** [comment]
Currently failing on this when i `pnpm next build` inside `examples/app-router`:
image

**Comment by khuezy** [comment]
Can you try reverting the changes to:
```
    "@types/react": "npm:types-react@19.0.0-rc.1",
    "@types/react-dom": "npm:types-react-dom@19.0.0-rc.1",
```

I'm using the 18.x "@types" in my app and it doesn't complain.

**Comment by sommeeeer** [comment]
> I'm using the 18.x `"@types"` in my app and it doesn't complain.

Alright, i did this instead: 
```json
"@types/react": "18.2.20",
"@types/react-dom": "18.2.7",```

seems to be coming further now. will look

**Comment by sommeeeer** [comment]
Not able to build `pages-router` after upgrading to next 15. I have tried both 19 and 18 for `@types/*`
image

**Comment by conico974** [comment]
@sommeeeer There is an issue with the lockfile https://github.com/opennextjs/opennextjs-aws/actions/runs/11466288277/job/31906625766?pr=588
Also it doesn't look like a typing issue`Check validity of types` did succeed.
Most likely it cannot find the correct version of react ( might be related to the lockfile issue )

**Comment by sommeeeer** [comment]
After installing `next@latest` in `pages-router` im running into this error: `Error: Objects are not valid as a React child (found: object with keys {$$typeof, type, key, ref, props, _owner, _store}). If you meant to render a collection of children, use an array instead.`. What happend with the validation? I would like to know if the lockfile is fine now.

**Comment by sommeeeer** [comment]
So I think its using the wrong dependency for React for some reason. If this is related to `pnpm` im not sure yet. However, I found out that building with this command works: `pnpx next build --experimental-build-mode compile`

**Comment by khuezy** [comment]
```--experimental-build-mode compile```
iirc, this flag prevents the prebuilding of the app.

**Comment by sommeeeer** [comment]
alright so im able to build every app now with `pnpm next build`. had to run the codemod on all the apps in `/examples` **and** on `/example`. choosing to upgrade react and react types to 19 aswell. after that i also had to update the react deps in `example/shared`. 

i also changed the worldtimeapi to be timeapi instead. (worldtimeapi blocked me after a while)

**Comment by sommeeeer** [comment]
so i was able to run `pnpm sst deploy --stage test` in `examples/sst` now. it hangs on this during `app-pages-router`: 
image

if you just wait it will finish in the end. and you get the deployed urls. as you can see the error is probably coming from a peer depedency of sst which is `ink`. its used in `sst/cli/deploy.js`. 

i fixed the e2e to work now and it passes 100%

File: packages/open-next/src/http/openNextResponse.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export interface StreamCreator {
 }
 
 // We only need to implement the methods that are used by next.js
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (sommeeeer) [reply packages/open-next/src/http/openNextResponse.ts:28]: Only way I could build `open-next`, else it was throwing errors on the class.

File: packages/open-next/CHANGELOG.md [hunk 2]
```diff
@@ -142,7 +142,7 @@
 
   This is the V3 of OpenNext. It includes some breaking changes and cannot be used as a drop-in replacement for V2. If your IAC is using OpenNext V2, you will need to update it to use V3.
 
-  If you are using OpenNext V2, please refer to the [migration guide](https://open-next.js.org/migration#from-opennext-v2) to upgrade to V3.
+  If you are using OpenNext V2, please refer to the [migration guide](https://opennext.js.org/migration#from-opennext-v2) to upgrade to V3.
```
[the remark this answers is not in the document]
Comment (khuezy) [reply packages/open-next/CHANGELOG.md:145]: `CHANGELOG.md` is auto generated. That might be just an old link. BTW the correct link is `https://opennext.js.org/aws/migration#from-opennext-v2`

[the remark this answers is not in the document]
Comment (sommeeeer) [reply packages/open-next/CHANGELOG.md:145]: This was a mistake. I will revert it.

File: packages/open-next/src/http/openNextResponse.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sommeeeer) [reply packages/open-next/src/http/openNextResponse.ts:28]: `Class 'OpenNextNodeResponse' incorrectly implements class 'ServerResponse<IncomingMessage>'. Did you mean to extend 'ServerResponse<IncomingMessage>' and inherit its members as a subclass?
  Property 'setHeaders' is missing in type 'OpenNextNodeResponse' but required in type 'ServerResponse<IncomingMessage>'.ts(2720)
http.d.ts(625, 9): 'setHeaders' is declared here.`

[the remark this answers is not in the document]
Comment (khuezy) [reply packages/open-next/src/http/openNextResponse.ts:28]: I checked out your branch and ran `npm run build`. 6/6 builds successful. Am I building the wrong thing?
My VSCode doesn't show any type errors on this class either.

[the remark this answers is not in the document]
Comment (sommeeeer) [reply packages/open-next/src/http/openNextResponse.ts:28]: > Oh wait there was the ts-ignore lol

yeah, we should resolve these before going further.

its expecting `setHeaders` on `OpenNextNodeResponse` class. Its required in `ServerResponse<IncomingMessage>`.  Should take in these params: `headers: Headers | Map<string, string | number | readonly string[]>`

[the remark this answers is not in the document]
Comment (khuezy) [reply packages/open-next/src/http/openNextResponse.ts:28]: that causes a bunch of additional type errors when you try to conform to the types: `number | readonly string[]`

[the remark this answers is not in the document]
Comment (sommeeeer) [reply packages/open-next/src/http/openNextResponse.ts:28]:
```ts
setHeaders(
  headers: Headers | Map<string, string | number | readonly string[]>,
): this {
  for (const [name, value] of Object.entries(headers)) {
    if (value !== undefined) {
      this.setHeader(name, value);
    }
  }
  return this;
}
```

[the remark this answers is not in the document]
Comment (khuezy) [reply packages/open-next/src/http/openNextResponse.ts:28]: `Object.entries(headers)` doesn't work unfortunately. `headers.entries()` returns the iterable. I think you'll need to do a typecheck on `Headers` vs `Map`

**Review by conico974 — APPROVED** [review]
Thanks for that @sommeeeer

**Comment by khuezy** [comment]
yay thanks @sommeeeer !

**Pull request merged**